ui/src/logger/logger.rs

```diff
@@ -0,0 +1,159 @@
+use once_cell::sync::Lazy;
+use regex::Regex;
+use std::fs::{File, OpenOptions};
+use std::io::{prelude::*, BufReader};
+use std::sync::Mutex;
+use warp::sync::RwLock;
+
+use chrono::Local;
+
+static LOG_ACTIVE: Lazy<RwLock<bool>> = Lazy::new(|| RwLock::new(false));
+
+static DEBUG_LOG_PATH: &str = ".uplink/debug.log";
+
+pub static LOGGER: Lazy<RwLock<Logger>> = Lazy::new(|| RwLock::new(Logger::load()));
+
+#[derive(Debug, Clone)]
+pub struct Log {
+    pub level: LogLevel,
+    pub message: String,
+    pub datetime: String,
+}
+
+#[derive(Debug, Clone)]
+pub enum LogLevel {
+    Debug,
+    Info,
+    Warn,
+    Error,
+}
+
+impl LogLevel {
+    pub fn from_str(log_level: &str) -> Self {
+        match log_level.trim() {
+            "Debug" => LogLevel::Debug,
+            "Warn" => LogLevel::Warn,
+            "Info" => LogLevel::Info,
+            "Error" => LogLevel::Error,
+            _ => LogLevel::Debug,
+        }
+    }
+
+    pub fn to_str(&self) -> &str {
+        match self {
+            LogLevel::Debug => "DEBUG",
+            LogLevel::Warn => "WARN",
+            LogLevel::Info => "INFO",
+            LogLevel::Error => "ERROR",
+        }
+    }
+
+    pub fn color(&self) -> &'static str {
+        match self {
+            LogLevel::Debug => "rgb(0, 255, 0)",
+            LogLevel::Info => "rgb(0, 195, 255)",
+            LogLevel::Warn => "yellow",
+            LogLevel::Error => "red",
+        }
+    }
+}
+
+pub struct Logger {
+    log_file: String,
+    log_entries: Mutex<Vec<Log>>,
+}
+
+impl Logger {
+    pub fn activate_logger() {
+        *LOG_ACTIVE.write() = true;
+    }
+
+    fn load() -> Logger {
+        let log_file = DEBUG_LOG_PATH.to_string();
+        let _ = OpenOptions::new().create(true).append(true).open(&log_file);
+        let log_entries = Mutex::new(Vec::new());
+        Logger {
+            log_file,
+            log_entries,
+        }
+    }
+}
+
+impl Logger {
+    fn log(&self, level: LogLevel, message: &str) {
+        if *LOG_ACTIVE.read() {
+            let new_log = Log {
+                level,
+                message: message.to_string(),
+                datetime: Local::now().to_string(),
+            };
+            let mut log_entries = self.log_entries.lock().unwrap();
+            log_entries.push(new_log.clone());
+
+            let mut file = OpenOptions::new()
+                .append(true)
+                .open(&self.log_file)
+                .unwrap();
+
+            if let Err(error) = writeln!(file, "{:?}", new_log.clone()) {
```

Comment:
I think it would be better to implement `Display` for `Log`. 
it's common for a logger implementation to display something like this: `<date time> <delimiter> <log level> <delimiter> <message>`.

 for example: `2022-01-19 | 09:41:30.33 | DEBUG | some message` 

another example:  `DEBUG | 2022-01-19 | 09:41:30.33 | module name | some message`